src/System.Data.Common/src/System/Data/Common/DbDataReaderExtensions.cs

```diff
@@ -2,22 +2,92 @@
+                BaseTableName = GetDbColumnValue<string>(SchemaTableColumn.BaseTableName);
+                ColumnName = GetDbColumnValue<string>(SchemaTableColumn.ColumnName);
+                ColumnOrdinal = GetDbColumnValue<int?>(SchemaTableColumn.ColumnOrdinal);
+                ColumnSize = GetDbColumnValue<int?>(SchemaTableColumn.ColumnSize);
+                IsAliased = GetDbColumnValue<bool?>(SchemaTableColumn.IsAliased);
+                IsAutoIncrement = GetDbColumnValue<bool?>(SchemaTableOptionalColumn.IsAutoIncrement);
+                IsExpression = GetDbColumnValue<bool>(SchemaTableColumn.IsExpression);
+                IsHidden = GetDbColumnValue<bool?>(SchemaTableOptionalColumn.IsHidden);
+                IsIdentity = GetDbColumnValue<bool?>("IsIdentity");
+                IsKey = GetDbColumnValue<bool?>(SchemaTableColumn.IsKey);
+                IsLong = GetDbColumnValue<bool?>(SchemaTableColumn.IsLong);
+                IsReadOnly = GetDbColumnValue<bool?>(SchemaTableOptionalColumn.IsReadOnly);
+                IsUnique = GetDbColumnValue<bool?>(SchemaTableColumn.IsUnique);
+                NumericPrecision = GetDbColumnValue<int?>(SchemaTableColumn.NumericPrecision);
+                NumericScale = GetDbColumnValue<int?>(SchemaTableColumn.NumericScale);
+                UdtAssemblyQualifiedName = GetDbColumnValue<string>("UdtAssemblyQualifiedName");
+                DataType = GetDbColumnValue<Type>(SchemaTableColumn.DataType);
+                DataTypeName = GetDbColumnValue<string>("DataTypeName");
+            }
+
+            private T GetDbColumnValue<T>(string columnName)
+            {
+                if (_schemaColumns.Contains(columnName) && _schemaRow[columnName] is T value)
+                {
+                    return value;
+                }
+                return default;
+            }
+        }
+
+        public static ReadOnlyCollection<DbColumn> GetColumnSchema(this DbDataReader reader)
         {
-            if (reader.CanGetColumnSchema())
+            if (reader is null)
+            {
+                throw new ArgumentNullException(nameof(reader));
+            }
+            if (reader is IDbColumnSchemaGenerator schemaGenerator)
             {
-                return ((IDbColumnSchemaGenerator)reader).GetColumnSchema();
+                return schemaGenerator.GetColumnSchema();
             }
-            throw new NotSupportedException();
+            return GetColumnSchemaCompatibility(reader);
         }
 
         public static bool CanGetColumnSchema(this DbDataReader reader)
         {
-            return reader is IDbColumnSchemaGenerator;
+            return true;
+        }
+
+        private static ReadOnlyCollection<DbColumn> GetColumnSchemaCompatibility(DbDataReader reader)
+        {
+            var columnSchema = new List<DbColumn>();
+            DataTable schemaTable = reader.GetSchemaTable();
```

Comment:
Some more vars missing

Reply:
The guidance I've been given (and seen others given) is that var can be used where the rhs of the expression repeats the type on the lhs. So line 81 is eligible, 82 isn't though because the rhs doesn't contain the type name. I makes things a little larger than it could be it does benefit readability when reviewing isolated snippets.

Reply:
OK, fair enough.

Reply:
Right, the consensus last time we discussed is that pretty much means in CoreFX `var` is fine iff the right side is either a constructor or it has an explicit cast.